src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((acc, item) => {
+    const groupKey = item[key] as unknown as string;
```

Comment:
Casting `item[key]` to `string` using `as unknown as string` might not be safe if `key` is not guaranteed to be a string. Consider using a more type-safe approach or ensure that `key` is always a string.